src/TestFramework/Microsoft.Rest.ClientRuntime.Azure.TestFramework/ConnectionStringKeys.cs

```diff
@@ -0,0 +1,126 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+namespace Microsoft.Rest.ClientRuntime.Azure.TestFramework
+{
+    using System;
+
+    /// <summary>
+    /// This class represents KeyNames for all the keys currently supported in Connection string    
+    /// !!!!!!ALERT!!!!!
+    /// If you add a public const field, it will be used/assumed as key as part of the connection string
+    /// This class is being used to reflect on all the key names supported in connection string.
+    /// </summary>
+    public class ConnectionStringKeys
+    {
+        // <summary>
+        /// The key inside the connection string for the management certificate
+        /// </summary>
+        public const string ManagementCertificateKey = "ManagementCertificate";
+
+        /// <summary>
+        /// The key inside the connection string for the subscription identifier
+        /// </summary>
+        public const string SubscriptionIdKey = "SubscriptionId";
+
+        /// <summary>
+        /// If a tenant other than common is to be used with the subscription, specifies the tenant
+        /// </summary>
+        public const string AADTenantKey = "AADTenant";
+
+        /// <summary>
+        /// The key inside the connection string for a Microsoft ID (OrgId or LiveId)
+        /// </summary>
+        public const string UserIdKey = "UserId";
+
+        /// <summary>
+        /// The key inside the connection string for a user password matching the Microsoft ID
+        /// </summary>
+        public const string PasswordKey = "Password";
+
+        /// <summary>
+        /// Service principal key
+        /// </summary>
+        public const string ServicePrincipalKey = "ServicePrincipal";
+
+        /// <summary>
+        /// The client ID to use when authenticating with AAD
+        /// </summary>
+        public const string AADClientIdKey = "AADClientId";
+
+        /// <summary>
+        /// ServicePrincipal Secret Key
+        /// </summary>
+        public const string ServicePrincipalSecretKey = "ServicePrincipalSecret";
+
+        /// <summary>
+        /// Environment name
+        /// </summary>
+        public const string EnvironmentKey = "Environment";
+
+        #region Tokens
+        /// <summary>
```

Comment:
Regions are meant for VS. So when the code do open in VS, we can use them. Keeping regions.

---
In reply to: [78469344](https://github.com/Azure/azure-sdk-for-net/pull/2328#discussion_r78469344) [](ancestors = 78469344)